app/controllers/record_controller.rb

```diff
@@ -22,6 +23,23 @@ def record
     render 'record'
   end
 
+  def direct_link
+    fetch_eds_record
+    redirect_to @record.fulltext_link[:url]
+  end
+
+  private
+
+  def valid_url!
+    return redirect_to root_url unless valid_url?
+  end
+
+  def restricted!
+    flash[:error] = 'Restricted access. Please use our feedback form for ' \
```

Comment:
This is a message we only want to show to guests, right? And in that case it would want to be inside the `if helpers.guest?` test. But as-is, wouldn't it actually display to everybody?

Reply:
I concur this feels weird as is. I think it ends up working just fine, because either the user is fine and they end up redirected to the target url, or they are not fine and redirect to the root with the flash message. However, it would be better to _not_ set the flash unless they are a guest accessing a restricted link so I'll tweak to remove confusion.

Reply:
👍